scheduler-server/src/main/java/gov/nasa/jpl/aerie/scheduler/server/services/SynchronousSchedulerAgent.java

```diff
@@ -440,20 +440,31 @@ private ScheduleResults collectResults(final Plan plan, final Map<ActivityInstan
       for (var goalEval : plan.getEvaluation().getGoalEvaluations().entrySet()) {
         var goalId = goalsToIds.get(goalEval.getKey());
         //goal could be anonymous, a subgoal of a composite goal for example, and thus have no meaning for results sent back
+        final var activitiesById = plan.getActivitiesById();
         if(goalId != null) {
           final var goalResult = new ScheduleResults.GoalResult(
               goalEval
                   .getValue()
-                  .getInsertedActivities()
-                  .stream()
-                  .map(instancesToIds::get)
-                  .collect(Collectors.toList()),
+                  .getInsertedActivities().stream().map(activityInstance -> {
+                    ActivityInstanceId id;
+                    if(activityInstance.isGenerated()){
+                      id = instancesToIds.get(activitiesById.get(activityInstance.getParentActivity()));
+                    } else{
+                      id = instancesToIds.get(activityInstance);
+                    }
+                    return id;
+                  }).toList(),
               goalEval
                   .getValue()
-                  .getAssociatedActivities()
-                  .stream()
-                  .map(instancesToIds::get)
-                  .collect(Collectors.toList()),
+                  .getAssociatedActivities().stream().map(activityInstance -> {
+                    ActivityInstanceId id;
+                    if(activityInstance.isGenerated()){
+                      id = instancesToIds.get(activitiesById.get(activityInstance.getParentActivity()));
+                    } else{
+                      id = instancesToIds.get(activityInstance);
+                    }
+                    return id;
+                  }).toList(),
```

Comment:
Can replace the `getInsertedActivities()` and `getAssociatedActivities()` logic with:
```java
.getAssociatedActivities().stream().map(activityInstance -> instancesToIds.get(activityInstance.isGenerated() ?
    activitiesById.get(activityInstance.getParentActivity()) :
    activityInstance))
```

Reply:
@adrienmaillard I've pushed to make `getParentActivity()` return an `Optional<SchedulingActivityInstanceId>`.

This makes client-side calls safes since they don't have to use `isGenerated()` guarding (which is not guaranteed to prevent `getParentActivity()` returning a `null` should `ActivityInstance`'s implementation change). Feel free to discard my commit, no hard feelings :)

Reply:
It looks better, thank you.